### DISCUSSION TITLE
Are (advisory) session locks supported?

### DISCUSSION THREAD

**Discussion by ariofrio** [porsager/postgres#387] [Q&A] [answered]
It's my understanding that because `postgres` doesn't expose connections/sessions, that it's not possible to hold a [session lock](https://www.postgresql.org/docs/14/explicit-locking.html#ADVISORY-LOCKS) for longer than one transaction. Is this accurate? For example:

```sql
select pg_advisory_lock(MY_APP_LOCK_ID)
```

**Comment by porsager** [comment, marked as answer]
Yes, you should be able to do that by creating a single connection instance like this:

```js
const sql = postgres({ max: 1 })
```

Would that be good enough?

**Reply by ariofrio** [reply]
Doesn't the connection get reset after each sql`` call?

**Reply by ariofrio** [reply]
I do need to use `listen()` as well. Would `postgres({max: 2})` work, then?

**Reply by porsager** [reply]
No, max specifies numbers of connections in the shared pool, and listen has it's own dedicated connection outside. 

Connections are only closed after max_lifetime (defaults to 45-90 mins) or after idle_timeout (default off), so you might want to run with `max_lifetime: false`
